Automattic/jetpack
# Social | Add `wpcom/v3/publicize/connections` endpoint

Extracted out of #40536

## Proposed changes:

* Add `wpcom/v3/publicize/connections` endpoint that can be used in every scenario the publicize module is used
    * WPCOM sites
    * Jetpack sites - both Jetpack and Social plugins

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







* Apply this PR to your sandbox and point public API to the sandbox
* Goto [Developer console](https://developer.wordpress.com/docs/api/console/)
* Hit WP REST API - `wpcom/v3/sites/<site_id>/publicize/connections`
* Confirm that the site connections are displayed correctly as per the schema in this PR
* Break a connection by removing Jetpack from the service, e.g. [here for Facebook](https://www.facebook.com/settings/?tab=business_tools).
* Add `?test_connections=1` to the endpoint above
* Verify that the tests are run for connections
* Verify that `status` is set to `'broken'` for the broken connection
* Point your Jetpack site to the sandbox
* Goto any page which uses `wp.apiFetch` e.g. Block Editor
* In console, run this
```js
await wp.apiFetch({
    path: '/wpcom/v3/publicize/connections'
});
```
* Confirm that the site connections are displayed correctly as per the schema in this PR

<img width="953" alt="Screenshot 2024-12-10 at 7 21 25 PM" src="https://github.com/user-attachments/assets/fe8fd883-b0c5-40b8-b204-660396fe92f8">

<img width="888" alt="Screenshot 2024-12-11 at 10 53 51 AM" src="https://github.com/user-attachments/assets/3f16fce7-8bba-4941-a995-8b7cda0b2803">


<img width="712" alt="Screenshot 2024-12-10 at 7 21 09 PM" src="https://github.com/user-attachments/assets/e0cc5910-6e0e-4bf3-b354-778d0ffee2a8">
